File: doc-examples/r2dbc-example-kotlin/src/main/kotlin/example/AuthorRepository.kt
package example

import io.micronaut.data.model.query.builder.sql.Dialect
import io.micronaut.data.r2dbc.annotation.R2dbcRepository
import io.micronaut.data.repository.reactive.ReactiveStreamsCrudRepository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import javax.validation.constraints.NotNull

@R2dbcRepository(dialect = Dialect.MYSQL) // <1>
interface AuthorRepository : ReactiveStreamsCrudRepository<Author, Long> {
    override fun findById(id: @NotNull Long): Mono<Author> // <2>
    override fun findAll(): Flux<Author>
}

File: doc-examples/r2dbc-example-python/src/main/python/example/AuthorRepository.py
from micronaut.data.model.query.builder.sql import Dialect
from micronaut.data.r2dbc.annotation import R2dbcRepository
from micronaut.data.repository.reactive import ReactiveStreamsCrudRepository
from reactor.core.publisher import Flux, Mono

from example.Author import Author


@R2dbcRepository(dialect=Dialect.POSTGRES)  # <1>
class AuthorRepository(ReactiveStreamsCrudRepository[Author, int]):

    def findById(self, id: int) -> Mono[Author]: ...  # <2>

    def findAll(self) -> Flux[Author]: ...
